cmd/compact_unix.go

```diff
@@ -0,0 +1,142 @@
+//go:build !windows
+// +build !windows
+
+/*
+ * JuiceFS, Copyright 2024 Juicedata, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"path/filepath"
+	"syscall"
+
+	"github.com/juicedata/juicefs/pkg/meta"
+	"github.com/juicedata/juicefs/pkg/utils"
+	"github.com/juicedata/juicefs/pkg/vfs"
+	"github.com/urfave/cli/v2"
+)
+
+func cmdCompact() *cli.Command {
+	return &cli.Command{
+		Name:      "compact",
+		Action:    compact,
+		Category:  "TOOL",
+		Usage:     "Trigger compaction of chunks",
+		ArgsUsage: "Path...",
+		Description: `
+ Examples:
+ # compact with path
+ $ juicefs compact /mnt/jfs/foo
+ `,
+		Flags: []cli.Flag{
```

Comment:
We can have a flag (`--min-slices`) to filter out files with fewer number of slices.

Reply:
i think that should add an arg minSliceCnt to engine.Compact func.
at line 2925, add some code like this, is it ok?
```
if len(ss) < minSliceCnt {
    return
}
```

https://github.com/juicedata/juicefs/blob/7b85645cac16c8442b3b7d35365a99dc0e339cdd/pkg/meta/redis.go#L2891-L2925